source/slang/slang-ir-use-uninitialized-values.cpp

```diff
@@ -434,13 +434,56 @@ namespace Slang
         return false;
     }
 
+    static bool isWrittenTo(IRInst* inst)
+    {
+        List<IRInst*> stores;
+        List<IRInst*> loads;
+
+        for (auto alias : getAliasableInstructions(inst))
+        {
+            for (auto use = alias->firstUse; use; use = use->nextUse)
+            {
+                IRInst* user = use->getUser();
+                collectLoadStore(stores, loads, user, alias);
+
+                // ...we will ignore the rest...
+                if (stores.getCount())
+                    return true;
+            }
+        }
+
+        return false;
+    }
+
+    static bool isDirectlyWrittenTo(IRInst* inst)
+    {
+        List<IRInst*> stores;
+        List<IRInst*> loads;
+
+        for (auto use = inst->firstUse; use; use = use->nextUse)
+        {
+            IRInst* user = use->getUser();
+            collectLoadStore(stores, loads, user, inst);
```

Comment:
This seems very inefficient. All we care about here is whether there are any stores, we don't want to collect all load and stores just to do the check.

Reply:
I've reworked the collection system so that now there are methods to check the usage type of an instruction wrt an operand. With this we can skip the collection and do a pure check.